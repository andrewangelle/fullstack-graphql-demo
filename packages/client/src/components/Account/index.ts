export * from './InventoryList'
export * from './PurchaseList';
export * from './AccountsPage';
export * from './SalesList';
export * from './UserInfo';
export * from './AccountsContainer'
export * from './AccountsPage'

export * from './Register';