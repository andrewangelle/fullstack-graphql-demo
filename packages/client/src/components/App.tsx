import React from 'react';
import { StripeProvider, Elements } from 'react-stripe-elements';

import {
  BcMenu,
  Routes,
} from 'components/'

const Stripe = StripeProvider as any

export function App(props: any) {
  return (
    <Stripe apiKey={process.env.REACT_APP_STRIPE_TOKEN}>
      <Elements>
        <BcMenu {...props}>
          <Routes {...props} />
        </BcMenu>
      </Elements>
    </Stripe>
  )
}
