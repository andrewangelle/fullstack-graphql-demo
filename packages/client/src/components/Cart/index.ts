export * from './PaymentPanel'
export * from './CartItems';
export * from './ItemsTable';
export * from './ItemQuantity';
export * from './RemoveIcon';
export * from './AddItemButton';
export * from './CartPage';
