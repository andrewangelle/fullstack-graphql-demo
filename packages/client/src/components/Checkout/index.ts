export * from './Payment';
export * from './CheckoutPage';

export * from './ShippingForm';
export * from './PaymentForm';
export * from './AuthorizeForm';