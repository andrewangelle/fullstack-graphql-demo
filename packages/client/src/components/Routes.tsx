
import React from 'react';
import { Router } from '@reach/router';

import {
  AccountsPage,
  CheckoutPage,
  LandingPage,
  LoginPage,
  LogoutPage,
  RegisterPage,
  ShopPage,
  ProductPage,
} from 'components/'

function Routes(props: any) {
  return (
    <Router>
      <LandingPage path="/" {...props} />
      <LoginPage path="/login" {...props} />
      <LogoutPage path="/logout" {...props} />
      <ProductPage path="/product" {...props} />
      <RegisterPage path="/register" {...props} />
      <AccountsPage path="/account" {...props} />
      <CheckoutPage path="/checkout" {...props} />
      <ShopPage path="/shop" {...props} />
    </Router>
  )
}

export { Routes }
