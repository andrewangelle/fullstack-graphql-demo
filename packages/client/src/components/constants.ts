export const wrapperStyles = {
  width: '50%',
  marginLeft: 'auto',
  marginRight: 'auto'
}

export const centerStyles = props => ({
  width: props.width || '50%',
  marginLeft: 'auto',
  marginRight: 'auto'
}) 

export const contentWrapperStyles = {
  margin: '24px 16px', 
  padding: 24, 
  background: '#fff', 
  minHeight: 800 
}

export const lightBlue = {color: '#1890ff'}

export const formItemLayout = {
  labelCol: { span: 4 },
  wrapperCol: { span: 14 },
}

export const pointer = {
  cursor: 'pointer'
}
export const flex = {
  display: 'inline-flex'
}
export const spacer = {
  margin: 'auto 0.5rem'
}


export const navMenu = {
  margin: ' 0 auto',
  display: 'flex',
  justifyContent: 'space-between',
  border: 'none'
}