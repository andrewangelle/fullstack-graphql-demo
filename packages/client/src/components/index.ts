export * from './constants';
export * from './shared';
export * from './Checkout';
export * from './Account';
export * from './Cart';
export * from './Shop'
export * from './Main';

export * from './Login/LoginForm';
export * from './Login/LoginPage';
export * from './Login/LogoutPage';

export * from './App'
export * from './Root'
export * from './Routes'