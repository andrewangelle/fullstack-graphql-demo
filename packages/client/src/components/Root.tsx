import React from 'react';
import { ApolloClient, NormalizedCacheObject } from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

import { App, Loader } from 'components/';
import { createClient } from 'store/'
import 'styles/index.css';

interface RootState {
  loaded: boolean;
  client: ApolloClient<NormalizedCacheObject> | null;
}

class Root extends React.Component<{}, RootState> {
  state = {
    loaded: false,
    client: null,
  }

  async componentDidMount() {
    const client = await createClient();
    await (window as any).Stripe
    this.setState({ client })
  }

  componentDidUpdate(prevState: RootState) {
    if (prevState.client !== this.state.client) {
      if (this.state.client && !this.state.loaded) {
        this.setState({ loaded: true })
      }
    }
  }

  render() {
    const { loaded, client } = this.state;

    if (!loaded) {
      return <Loader />
    }

    return client !== null && loaded && (
      <ApolloProvider client={client}>
        <App {...this.state} />
      </ApolloProvider>
    )

  }
}


export { Root }