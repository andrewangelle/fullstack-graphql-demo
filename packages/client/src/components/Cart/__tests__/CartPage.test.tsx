import React from 'react';
import { InMemoryCache } from 'apollo-boost';
import wait from 'waait';

import { CartPage } from 'components/';
import { mockApollo, cartMock } from 'test-utils';

describe('CartPage', () => {
  it('renders cart page', async () => {
    const component = mockApollo(<CartPage toggleCart={() => null} />, {
      mocks: [cartMock],
      cache: new InMemoryCache()
    });
    await wait(0)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
