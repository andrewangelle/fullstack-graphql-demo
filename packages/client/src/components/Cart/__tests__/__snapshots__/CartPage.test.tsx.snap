// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage renders cart page 1`] = `
<section
  className="ant-layout ant-layout-has-sider"
>
  <main
    className="ant-layout-content"
  >
    <div
      style={
        Object {
          "minHeight": "auto",
        }
      }
    >
      <div
        className="ant-table-wrapper"
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
              style={Object {}}
            >
              <div
                className="ant-table-content"
              >
                <div
                  className="ant-table-body"
                  onScroll={[Function]}
                  onWheel={[Function]}
                  style={Object {}}
                >
                  <table
                    className=""
                    style={Object {}}
                  >
                    <colgroup>
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": undefined,
                            "width": undefined,
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr
                        style={
                          Object {
                            "height": null,
                          }
                        }
                      >
                        <th
                          className=""
                        >
                          <div>
                            Name
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div>
                            Quantity
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div>
                            Price
                          </div>
                        </th>
                        <th
                          className=""
                        >
                          <div>
                            Remove
                          </div>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  className="ant-table-placeholder"
                >
                  <div
                    className="ant-empty ant-empty-normal"
                  >
                    <div
                      className="ant-empty-image"
                    >
                      <img
                        alt="No Data"
                        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                      />
                    </div>
                    <p
                      className="ant-empty-description"
                    >
                      No Data
                    </p>
                  </div>
                </div>
                <div
                  className="ant-table-footer"
                >
                  <div
                    className="ant-row"
                  >
                    <div
                      className="ant-col-9 ant-col-push-18"
                    >
                      <h4>
                        Total: $
                        0
                      </h4>
                      <button
                        className="ant-btn ant-btn-ghost"
                        onClick={[Function]}
                        type="button"
                      >
                        <a>
                           Clear Cart
                        </a>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <header
        className="ant-layout-header"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <h4
          style={
            Object {
              "color": "white",
            }
          }
        >
          Total: $
          0
        </h4>
        <button
          className="ant-btn ant-btn-default"
          onClick={[Function]}
          type="button"
        >
          <span>
            Checkout
          </span>
        </button>
      </header>
    </div>
  </aside>
</section>
`;
